cluster/src/main/java/io/scalecube/cluster/ClusterConfig.java

```diff
@@ -268,13 +278,17 @@ public String toString() {
   }
 
   private String metadataAsString() {
-    return metadata.isEmpty() ? "[]" : "metadata@" + Integer.toHexString(metadata.hashCode());
+    return metadata.remaining() == 0
+        ? "[]"
+        : "metadata@" + Integer.toHexString(metadata.hashCode()) + "[" + metadata.remaining() + "]";
   }
 
   public static final class Builder {
 
+    public static final ByteBuffer EMPTY = ByteBuffer.wrap(new byte[0]);
+
     private List<Address> seedMembers = Collections.emptyList();
-    private Map<String, String> metadata = new HashMap<>();
+    private ByteBuffer metadata = EMPTY;
```

Comment:
IMHO `ClusterConfig`'s (further `CF`)  `metadata`  has nothing to do with ByteBuffer, it's  `MetadataCodec`  must be concerned about encoding CF.metadata to advertise it to the cluster. BTW, for local CF metadata only process of converting `Object` -> `Bytes` is relevant.  For a part where we getting remote metadata then decoding process (`Bytes` -> `Object`  )  is taking place.